File: problems/0078-subsets/python/solution.py
def subsets(nums):
    """Subsets — https://leetcode.com/problems/subsets/"""
    res = []
    subset = []

    def backtrack(i):
        if i == len(nums):
            res.append(subset[:])
            return
        subset.append(nums[i])  # include nums[i]
        backtrack(i + 1)
        subset.pop()            # exclude nums[i]
        backtrack(i + 1)

    backtrack(0)
    return res

File: problems/0078-subsets/typescript/solution.ts
// Subsets — https://leetcode.com/problems/subsets/
export function subsets(nums: number[]): number[][] {
  const res: number[][] = [];
  const subset: number[] = [];
  const backtrack = (i: number): void => {
    if (i === nums.length) {
      res.push([...subset]);
      return;
    }
    subset.push(nums[i]); // include
    backtrack(i + 1);
    subset.pop(); // exclude
    backtrack(i + 1);
  };
  backtrack(0);
  return res;
}
